### PR TITLE
Tryout

### PR DESCRIPTION
**Pull request by jelly**

### CODE REVIEW

File: test/check-application [hunk 1]
```diff
@@ -78,8 +78,7 @@ class TestApplication(testlib.MachineCase):
         self.admin_s.execute("systemctl --user stop podman.service; systemctl --now --user enable podman.socket")
         self.restore_dir("/home/admin/.local/share/containers", reboot_safe=True)
         self.addCleanup(self.admin_s.execute, "systemctl --user stop podman.service podman.socket || true")
-        self.addCleanup(self.admin_s.execute, "podman rm --force --all || true")
-        self.addCleanup(self.admin_s.execute, "podman pod rm --force --all || true")
+        self.addCleanup(self.admin_s.execute, "podman system reset --force")
```
Comment (martinpitt) [inline test/check-application:93]: But this also removes all images, and we want/need to keep them.

File: test/check-application [hunk 2]
```diff
@@ -60,6 +60,15 @@ class TestApplication(testlib.MachineCase):
             findmnt --list -otarget | grep /var/lib/containers/. | xargs -r umount
             """)
 
+        # copy images for user podman tests; podman insists on user session
+        m.execute("""
+            loginctl enable-linger $(id -u admin)
+            for img in quay.io/libpod/busybox quay.io/libpod/alpine quay.io/cockpit/registry:2; do
```
Comment (martinpitt) [inline test/check-application:70]: It's a bit awkward to have to repeat the container list here. Could we not just restore the tarballs into both system and user, or iterate over `podman images -n` or so?

File: test/check-application [hunk 3]
```diff
@@ -76,7 +85,6 @@ class TestApplication(testlib.MachineCase):
 
         # Enable user service as well
         self.admin_s.execute("systemctl --user stop podman.service; systemctl --now --user enable podman.socket")
-        self.restore_dir("/home/admin/.local/share/containers", reboot_safe=True)
```
Comment (martinpitt) [inline test/check-application:79]: Some form of this is still necessary. Does `podman system reset` do that? (I suppose it does)

Note that it's *not* the user dir which causes trouble, it's the system dir.

File: test/vm.install [hunk 4]
```diff
@@ -35,12 +35,15 @@ if [ "$ID" = "fedora" ]; then
     fi
 fi
 
-# copy images for user podman tests; podman insists on user session
-loginctl enable-linger $(id -u admin)
+# Store pristine containers so they can be restored.
+imagesdir="/var/lib/cockpittest/podman"
```
Comment (martinpitt) [inline test/vm.install]: testlib uses /var/lib/cockpittest/, please use a different directory.

File: test/vm.install [hunk 5]
```diff
@@ -36,7 +36,7 @@ if [ "$ID" = "fedora" ]; then
 fi
 
 # Store pristine containers so they can be restored.
-imagesdir="/var/lib/cockpittest/podman"
+imagesdir="/var/cache/podmantest"
```
Comment (martinpitt) [inline test/vm.install:39]: Ah, exactly -- but please squash this.

File: test/check-application (hunk 1 above)
Comment (martinpitt) [reply test/check-application:93]: .. I see, a subsequent commit then does that image restoration. This needs some squashing.

**Comment by martinpitt** [comment]
Hmm, [this failure](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1043-20220719-110835-c36986a6-fedora-36/log.html#9-2) does not raise my confidence..
```
time="2022-07-19T11:17:58Z" level=error msg="unlinkat /home/admin/.local/share/containers/storage: directory not empty"
Traceback (most recent call last):
  File "/usr/lib64/python3.10/unittest/case.py", line 555, in _callCleanup
    function(*args, **kwargs)
  File "/work/bots/make-checkout-workdir/bots/machine/machine_core/ssh_connection.py", line 368, in execute
    raise subprocess.CalledProcessError(proc.returncode, command, output=output)
subprocess.CalledProcessError: Command 'podman system reset --force' returned non-zero exit status 125.
```

**Comment by jelly** [comment]
> Hmm, [this failure](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1043-20220719-110835-c36986a6-fedora-36/log.html#9-2) does not raise my confidence..
> 
> ```
> time="2022-07-19T11:17:58Z" level=error msg="unlinkat /home/admin/.local/share/containers/storage: directory not empty"
> Traceback (most recent call last):
>   File "/usr/lib64/python3.10/unittest/case.py", line 555, in _callCleanup
>     function(*args, **kwargs)
>   File "/work/bots/make-checkout-workdir/bots/machine/machine_core/ssh_connection.py", line 368, in execute
>     raise subprocess.CalledProcessError(proc.returncode, command, output=output)
> subprocess.CalledProcessError: Command 'podman system reset --force' returned non-zero exit status 125.
> ```

I guess the fix you did by waiting till podman was really killed was the real fix.

**Pull request closed without merging**